packages/nuxt/src/pages/utils.ts

```diff
@@ -140,19 +140,19 @@ export function generateRoutesFromFiles (files: ScannedFile[], options: Generate
   return prepareRoutes(routes)
 }
 
-export async function augmentPages (routes: NuxtPage[], vfs: Record<string, string>, augmentedPages = new Set<string>()) {
+export async function augmentPages (routes: NuxtPage[], vfs: Record<string, string>, augmentedPagesWithPaths = new Set<string>()) {
   for (const route of routes) {
-    if (route.file && !augmentedPages.has(route.file)) {
+    if (route.file && !augmentedPagesWithPaths.has(route.file)) {
       const fileContent = route.file in vfs ? vfs[route.file] : fs.readFileSync(await resolvePath(route.file), 'utf-8')
       Object.assign(route, await getRouteMeta(fileContent, route.file))
-      augmentedPages.add(route.file)
+      augmentedPagesWithPaths.add(`${route.file}_${route.path}`)
```

Comment:
if we are doing this, we should probably build the path from parents as well, as a child path might otherwise cause a false positive match.

e.g. parent with path `/test` and a child with path `foo` would cause a match with parent with path `/other` and a child with path `foo`, if they were the same filename....

Reply:
Good point

Reply:
Added a `parentPath` param for recursive calls